admin/spec/front-end/pupil-access-arrangements-selection.spec.js

```diff
@@ -0,0 +1,188 @@
+    '        </div>' +
+    '        <div class="hide-checkbox-content">' +
+    '            <div class="form-group">' +
+    '' +
+    '                <div class="multiple-choice">' +
+    '                    <input id="questionReaderReason-0" type="radio" name="questionReaderReason"' +
+    '                           class="question-reader-reason" value="EAL">' +
+    '                    <label for="questionReaderReason-0">English as an additional language (EAL)</label>' +
+    '                </div>' +
+    '                <div class="multiple-choice">' +
+    '                    <input id="questionReaderReason-1" type="radio" name="questionReaderReason-1"' +
+    '                           class="question-reader-reason" value="SLP">' +
+    '                    <label for="questionReaderReason-1">Slow processing</label>' +
+    '                </div>' +
+    '                <div class="multiple-choice">' +
+    '                    <input id="questionReaderReason-2" type="radio" name="questionReaderReason-2"' +
+    '                           class="question-reader-reason" value="VIM">' +
+    '                    <label for="questionReaderReason-2">Visual impairments</label>' +
+    '                </div>' +
+    '                <div class="multiple-choice">' +
+    '                    <input id="questionReaderReason-3" type="radio" name="questionReaderReason-3"' +
+    '                           class="question-reader-reason" value="OTH" >' +
+    '                    <label for="questionReaderReason-3">Other</label>' +
+    '                </div>' +
+    '                <div class="panel panel-border-wide js-hidden">' +
+    '                    <div class="form-group">' +
+    '                        <div class="form-label">Please explain why the pupil needs this arrangement</div>' +
+    '                        <textarea id="questionReaderOtherInformation" name="questionReaderOtherInformation"' +
+    '                                  class="form-control form-control-3-4 restart-reason-info" rows="3"' +
+    '                                  maxlength="1000"></textarea>' +
+    '                    </div>' +
+    '                </div>' +
+    '                <br>' +
+    '            </div>' +
+    '        </div>' +
+    '    </li>' +
+    '    <li>' +
+    '        <div class="font-small">' +
+    '            <label for="accessArrangement-6">' +
+    '                Remove on-screen number pad' +
+    '            </label>' +
+    '        </div>' +
+    '        <div class="multiple-choice-mtc">' +
+    '            <input id="accessArrangement-6" name="accessArrangement[6]" type="checkbox" value="RON"' +
+    '                   aria-label="Tick accessArrangement Remove on-screen number pad." aria-checked="false"' +
+    '                   role="checkbox">' +
+    '            <div></div>' +
+    '        </div>' +
+    '    </li>')
+
+  $accessArrangementsList.append($accessArrangementsListItems)
```

Comment:
if you were to make the `pupil-access-arrangements-selection.js` injectable so that `window.GOVUK` could be mocked, would all this html be necessary?  also, could you not build up the DOM programatically instead of having all the mark up?

Reply:
@GuyHarwood field type is change to `recordedBy_user_id ` and test's DOM is being generated instead of statically declared.